examples/python/builtin_rsa.py

```diff
@@ -0,0 +1,93 @@
+                      AuthenticationClient, FrontendClient, FunctionInput,
+                      FunctionOutput, OwnerList, DataMap)
+from utils import (AUTHENTICATION_SERVICE_ADDRESS, FRONTEND_SERVICE_ADDRESS,
+                   AS_ROOT_CA_CERT_PATH, ENCLAVE_INFO_PATH, USER_ID,
+                   USER_PASSWORD)
+
+
+class BuiltinRsaExample:
+    def __init__(self, user_id, user_password):
+        self.user_id = user_id
+        self.user_password = user_password
+
+    def rsa(self):
+        client = AuthenticationService(
+            AUTHENTICATION_SERVICE_ADDRESS, AS_ROOT_CA_CERT_PATH,
+            ENCLAVE_INFO_PATH).connect().get_client()
+
+        print("[+] registering user")
+        client.user_register(self.user_id, self.user_password)
+
+        print("[+] login")
+        token = client.user_login(self.user_id, self.user_password)
+
+        client = FrontendService(FRONTEND_SERVICE_ADDRESS,
+                                 AS_ROOT_CA_CERT_PATH,
+                                 ENCLAVE_INFO_PATH).connect().get_client()
+        metadata = {"id": self.user_id, "token": token}
+        client.metadata = metadata
+
+        print("[+] registering function")
+        function_id = client.register_function(
+            name="builtin-rsa",
+            description="Native Rsa Encrypt Function",
+            executor_type="builtin",
+            payload=list(b"test data"),
+            inputs=[FunctionInput("rsa_key", "Input key file.")])
+
+        print("[+] registering input file")
+        """
+        Commands when encrypting input files:
+            ./teaclave_cli encrypt 
+                --algorithm teaclave-file-128 
+                --input-file ./tests/fixtures/functions/rsa/key.der 
+                --key 00000000000000000000000000000003 
+                --output-file ./tests/fixtures/functions/rsa/rsakey.enc 
+                --cmac-flag
+        """
+        url = "http://localhost:6789/fixtures/functions/rsa/rsakey.enc"
+        cmac = "4de3bb77327c82923640835c6e5ada66"
+        schema = "teaclave-file-128"
+        key = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3]
+        iv = []
+        key_data_id = client.register_input_file(url, schema, key, iv,
+                                                      cmac)
+
+        print("[+] creating task")
+        task_id = client.create_task(
+            function_id=function_id,
+            executor="builtin",
+            function_arguments=({}),
+            inputs_ownership=[OwnerList("rsa_key", [self.user_id])])
+
+        print("[+] assigning data to task")
+        client.assign_data_to_task(
+            task_id, [DataMap("rsa_key", key_data_id)], [])
+
+        print("[+] approving task")
+        client.approve_task(task_id)
+
+        print("[+] invoking task")
+        client.invoke_task(task_id)
+
+        print("[+] getting result")
+        result = client.get_task_result(task_id)
+        print("[+] done")
+
+        return bytes(result)
+
+
+def main():
```

Comment:
Is there any real world scenario for a single user to use this function? It might make more sense for two party.